domains/transfer/Transfer.API/API/Data/ConnectionRepository.cs

```diff
@@ -16,6 +16,31 @@ public ConnectionRepository(ApplicationDbContext context)
         this.context = context;
     }
 
+    public async Task AddConnectionAndDeleteInvitation(Connection newConnection, Guid invitationId)
+    {
+        await using var transaction = await context.Database.BeginTransactionAsync();
+
+        try
+        {
+            await context.Connections.AddAsync(newConnection);
+
+            var invitation = await context.ConnectionInvitations.FindAsync(invitationId);
```

Comment:
Why not FirstOrDefault(x => x.Id == invitationId)?

Reply:
FindAsync, checks the EF Core cache, before trying to query the database. That is why we use FindAsync. Since we are accessing the POST connection endpoint through accessing the GET connection-invitation endpoint, we can be pretty sure that the invitation is existing in the cache, when trying to add a new Connection to the databaser.